### PR TITLE
Let the JSON reader recover from unexpected inputs

### PR DESCRIPTION
**Pull request by tobim**
###  :memo: Checklist

- [x] All user-facing changes have changelog entries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ This changelog documents all notable user-facing changes of VAST.
 
 ## Unreleased
 
+- 🐞 The new simdjson based JSON reader introduced in
+  [#1356](https://github.com/tenzir/vast/pull/1356) is now more robust when
+  dealing with malformed inputs.
+  [#1386](https://github.com/tenzir/vast/pull/1386)
```
Comment (mavam) [inline CHANGELOG.md:19]: Please explain the bug more clearly.

File: libvast/vast/format/json.hpp [hunk 2]
```diff
@@ -210,8 +210,16 @@ caf::error reader<Selector>::read_impl(size_t max_events, size_t max_slice_size,
     if (bptr == nullptr)
       return caf::make_error(ec::parse_error, "unable to get a builder");
     if (auto err = add(*bptr, get_object_result.value(), *layout)) {
-      err.context() += caf::make_message("line", lines_->line_number());
-      return finish(cons, err);
+      if (err == ec::convert_error) {
+        if (num_invalid_lines_ == 0)
+          VAST_WARN("{} failed to convert value(s) in line {} : {}",
```
Comment (dominiklohmann) [inline libvast/vast/format/json.hpp:215]: While you're touching this code, can you remove the space after the line number? Might have other occurrences as well.

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,12 @@ This changelog documents all notable user-facing changes of VAST.
 
 ## Unreleased
 
+- 🐞 A Bug in the new simdjson based JSON reader introduced in
```
Comment (dominiklohmann) [inline CHANGELOG.md:16]:
```suggestion
- 🐞 A bug in the new simdjson based JSON reader introduced in
```

**Review by dominiklohmann — APPROVED** [review]
I tested this locally with the example you gave me via Slack (see below, `write_ts` in line 2) and was able to reproduce the faulty behavior on master and the warning with this PR.

```json
{"_path":"sip","_system_name":"sensor_10_0_0_2","_write_ts":"2021-02-01T13:23:04.768321Z","ts":"2021-02-01T13:18:04.777419Z","uid":"C6j3vyC2g0ND18JBl","id.orig_h":"217.182.199.128","id.orig_p":5132,"id.resp_h":"194.94.128.27","id.resp_p":5060,"trans_depth":0,"method":"OPTIONS","uri":"sip:100@194.94.128.27","request_from":"\"sipvicious\"<sip:100@1.1.1.1>","request_to":"\"sipvicious\"<sip:100@1.1.1.1>","call_id":"630386036184203546741250","seq":"1 OPTIONS","request_path":["SIP/2.0/UDP 217.182.199.128:5132"],"response_path":[],"user_agent":"friendly-scanner","request_body_len":0}
{"_path":"sip","_system_name":"sensor_10_0_0_2","_write_ts":"2021-02-01T13:2public","get_requests":2,"get_bulk_requests":0,"get_responses":0,"set_requests":0}
```

**Pull request merged**